packages/php-wasm/compile/build-assets/php_wasm.c

```diff
@@ -278,21 +295,19 @@ PHP_FUNCTION(post_message_to_js)
 		return;
 	}
 
-	char *result = js_module_onMessage(data);
-
-	if(result != NULL) {
-		// In PHP 7, the second parameter was removed and the string is always
-		// returned as a copy.
-		#if PHP_MAJOR_VERSION >= 7
-			RETURN_STRING(result);
-		#else
-			RETURN_STRING(result, 1);
-		#endif
-	} else {
+	char *response;
+	size_t response_len = js_module_onMessage(data, &response);
```

Comment:
Neat! I'd have to figure out how to allocate structs on the JavaScript side, but AFAIR that's just a few pointers next to each other? So maybe this would work:

```js
const structSize = 1 /* did_succeed */ + 4 /* *data */ + 4 /* data_len */;
const structPtr = _malloc(responseSize);
HEAPU8[structPtr] = didSucceed ? 0 : 1;
HEAPU8[structPtr + 1] = dataPtr;
HEAPU8[structPtr + 2] = dataPtr >> 8;
HEAPU8[structPtr + 3] = dataPtr >> 16;
HEAPU8[structPtr + 4] = dataPtr >> 24;
HEAPU8[structPtr + 5] = dataSize;
HEAPU8[structPtr + 6] = dataSize >> 8;
HEAPU8[structPtr + 7] = dataSize >> 16;
HEAPU8[structPtr + 8] = dataSize >> 24;
```

Rebuilding PHP is a bit tedious so I might get this one in without a struct, but I would definitely want to make working with structs a part of Playground.